torchtune/utils/_generation.py

```diff
@@ -4,141 +4,115 @@
 
-    # sample the next token
-    token = multinomial_sample_one(probs)
-    return token
-
-
-def generate_next_token(
-    model: TransformerDecoder,
-    input_pos: torch.Tensor,
-    x: torch.Tensor,
-    temperature: float = 1.0,
-    top_k: Optional[int] = None,
-) -> torch.Tensor:
-    # x: [1, s]
-    # input_pos: [s]
-    logits = model(x, input_pos)
-
-    # logits: [1, s, v] where v is vocab_size
-    # for sampling we extract the logits for the
-    # last token and convert to shape: [v]
-    logits = logits[0, -1]
 
-    # sample the next token
-    token = sample(logits, temperature, top_k)
-    return token
+def generate_next_token(model, input_pos, x, temperature=1.0, top_k=None):
+    """Generates the next token."""
+    logits = model(x, input_pos)[:, -1]
+    return sample(logits, temperature, top_k)
 
 
 @torch.inference_mode()
 def generate(
     model: TransformerDecoder,
     prompt: torch.Tensor,
+    *,
     max_generated_tokens: int,
     temperature: float = 1.0,
     top_k: Optional[int] = None,
-    stop_tokens: Optional[Set[int]] = None,
+    stop_tokens: Optional[List[int]] = None,
     custom_generate_next_token: Optional[Callable] = None,
 ) -> torch.Tensor:
     """
-    Generate tokens from a model conditioned on a prompt.
+    Generates tokens from a model conditioned on a prompt.
 
     Args:
         model (TransformerDecoder): model used for generation
-        prompt (torch.Tensor): tensor with the token IDs associated with the given
-            prompt. This is the output of the relevant tokenizer
-        max_generated_tokens (int): number of tokens to be generated. This is the max
-            since we can stop early based on whether the eos token is respected or not
-        temperature (float): value to scale the predicted logits by. Default is 1.0
-        top_k (Optional[int]): If specified, we prune the sampling to only token ids within
-            the top_k probabilities. Default is None
-        stop_tokens (Optional[Set[int]]): If specified, generation is stopped when any of these
-            tokens are generated. Default: None
-        custom_generate_next_token (Optional[Callable]): If specified, we'll use the custom
-            generate_next_token function (e.g. compiled function) when generating the tokens,
-            otherwise we'll use the default `geenrate_next_token` function. Default is None
+        prompt (torch.Tensor): tensor with the token IDs associated with the given prompt
+        max_generated_tokens (int): number of tokens to be generated
+        temperature (float): value to scale the predicted logits by
+        top_k (Optional[int]): If specified, we prune the sampling to only token ids within the top_k probabilities
+        stop_tokens (Optional[List[int]]): If specified, generation is stopped when any of these tokens are generated
+        custom_generate_next_token (Optional[Callable]): If specified, we'll use the custom generate_next_token function
```

Comment:
Can we clarify this is used for compiled version of the ```generate_next_token```? I think several folks would have that confusion.